jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +66,88 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the file.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath) throws IOException {
+    TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider();
+    Path tempDirectoryPath = tempDirectoryProvider.newDirectory();
+    ZipUtil.unzip(jarPath, tempDirectoryPath);
+
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isResourceFile = path -> !path.getFileName().toString().endsWith(".class");
+
+    // Determine class and resource files in the directory containing jar contents and create
+    // FileEntriesLayer for each type of layer (class or resource), while maintaining the
+    // file's original project structure.
+    FileEntriesLayer classesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                tempDirectoryPath,
+                isClassFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Classes")
+            .build();
+    FileEntriesLayer resourcesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                tempDirectoryPath,
+                isResourceFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Resources")
+            .build();
+
+    // Get dependencies from Class-Path in the jar's manifest and create a FileEntriesLayer.Builder
+    // with these dependencies as entries.
+    List<Path> dependencies = new ArrayList<>();
+    JarFile jarFile = new JarFile(jarPath.toFile());
+    String classPath = jarFile.getManifest().getMainAttributes().getValue("Class-Path");
+    if (classPath != null) {
+      for (String dependency : Splitter.on(" ").split(classPath)) {
+        Path depPath = Paths.get(dependency);
+        dependencies.add(depPath);
+      }
+    } else {
+      throw new IllegalStateException("Class path is not specified.");
```

Comment:
Do we have to fail here? Would this code path also handle the shaded jar style jars (which may not have `Class-Path` entries)?

Reply:
That's a good question. I didn't intend for this code path to handle shaded jars. It mainly focuses on regular jars, which require the class-path to be fully specified in order for it to work. We could try to intelligently determine what the dependencies are in the future implementations, but for now I decided to first check that the Class-Path was present before proceeding to the next steps. What do you think?

Reply:
A regular jar includes one that doesn't have Class-Path. It's possible that many people will try a hello-world app (i.e, no dependencies), so I think we shouldn't fail.

Reply:
Oh that's a very valid point! Okay I will remove this exception.